### PR TITLE
adding support for KML Styles

### PR DESCRIPTION
**Pull request by ldanzinger**
@JamesMBallard please review

Now that we have KmlStyle for 100.7, this can be added to the sample to closer match the common design

QML sample also was restructured slightly to make the code more declarative

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h [hunk 1]
```diff
@@ -68,16 +69,22 @@ class CreateAndSaveKmlFile : public QObject
...
 
   bool m_busy = false;
 
   Esri::ArcGISRuntime::Geometry createPoint() const;
   Esri::ArcGISRuntime::Geometry createPolyline() const;
   Esri::ArcGISRuntime::Geometry createPolygon() const;
   Esri::ArcGISRuntime::Geometry createEnvelope() const;
+  Esri::ArcGISRuntime::KmlStyle* createPointStyle();
+  Esri::ArcGISRuntime::KmlStyle* createLineStyle();
+  Esri::ArcGISRuntime::KmlStyle* createPolygonStyle();
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h]: Consider renaming these methods. A KmlPolygonStyle is not a KmlStyle. A KmlStyle has-a KmlPolygonStyle. The method names could add to confusion.

How about these?
```c++
  Esri::ArcGISRuntime::KmlStyle* createKmlStyleWithPointStyle();
  Esri::ArcGISRuntime::KmlStyle* createKmlStyleWithLineStyle();
  Esri::ArcGISRuntime::KmlStyle* createKmlStyleWithPolygonStyle();
```

They're wordier, but I think they better communicate the object model and what's happening.

File: ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h [hunk 2]
```diff
@@ -68,16 +69,22 @@ class CreateAndSaveKmlFile : public QObject
   Esri::ArcGISRuntime::Point m_point;
   Esri::ArcGISRuntime::Polyline m_polyline;
   Esri::ArcGISRuntime::Polygon m_polygon;
+  Esri::ArcGISRuntime::KmlStyle* m_pointStyle = nullptr;
+  Esri::ArcGISRuntime::KmlStyle* m_lineStyle = nullptr;
+  Esri::ArcGISRuntime::KmlStyle* m_polygonStyle = nullptr;
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h]: Consider renaming these for clarity. The object model isn't intuitive, so I would be overly descriptive on the names to help users.

```c++
  Esri::ArcGISRuntime::KmlStyle* m_kmlStyleWithPointStyle = nullptr;
  Esri::ArcGISRuntime::KmlStyle* m_kmlStyleWithLineStyle = nullptr;
  Esri::ArcGISRuntime::KmlStyle* m_kmlStyleWithPolygonStyle = nullptr;
```

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.qml [hunk 3]
```diff
@@ -129,40 +131,64 @@ Rectangle {
         }
     }
 
+    KmlStyle {
+        id: pointStyle
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.qml]:
```suggestion
        id: styleWithPointStyle
```

File: ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/CreateAndSaveKmlFile/CreateAndSaveKmlFile.h]: Don't think so. Results in build errors

**Comment by ldanzinger** [comment]
@JamesMBallard great points. Updated with your suggestions to better reflect the object model

**Pull request merged**